Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+    //    {
+    //        var settings = AssetDatabase.LoadAssetAtPath<ScriptReferenceRetargettingSettings>(k_MyCustomSettingsPath);
+    //        if (settings == null)
+    //        {
+    //            settings = CreateInstance<ScriptReferenceRetargettingSettings>();
+    //            settings.m_Number = 42;
+    //            settings.m_SomeString = "The answer to the universe";
+    //            AssetDatabase.CreateAsset(settings, k_MyCustomSettingsPath);
+    //            AssetDatabase.SaveAssets();
+    //        }
+    //        return settings;
+    //    }
+
+    //    internal static SerializedObject GetSerializedSettings()
+    //    {
+    //        return new SerializedObject(GetOrCreateSettings());
+    //    }
+    //}
+
+    //internal static class ScriptReferenceRetargettingSettingsIMGUIRegister
+    //{
+    //    [SettingsProvider]
+    //    public static SettingsProvider CreateMyCustomSettingsProvider()
+    //    {
+    //        SettingsProvider provider = new SettingsProvider("Project/ScriptReferenceRetargetter", SettingsScope.Project)
+    //        {
+    //            // By default the last token of the path is used as display name if no label is provided.
+    //            label = "Script Reference Retargetter Settings",
+    //            // Create the SettingsProvider and initialize its drawing (IMGUI) function in place:
+    //            guiHandler = (searchContext) =>
+    //            {
+    //                var settings = ScriptReferenceRetargettingSettings.GetSerializedSettings();
+    //                EditorGUILayout.PropertyField(settings.FindProperty("m_Number"), new GUIContent("My Number"));
+    //                EditorGUILayout.PropertyField(settings.FindProperty("m_SomeString"), new GUIContent("My String"));
+    //            },
+
+    //            // Populate the search keywords to enable smart search filtering and label highlighting:
+    //            keywords = new HashSet<string>(new[] { "Number", "Some String" })
+    //        };
+
+    //        return provider;
+    //    }
+    //}
+
+
+    public static class AssetScriptReferenceRetargeter
+    {
+        private struct ClassInformation
+        {
+            public string Name;
+            public string Namespace;
+            public string Guid;
+            public long FileId;
+        }
+
+        private const string YamlPrefix = "%YAML 1.1";
+
+        private static readonly Dictionary<string, string> sourceToOutputFolders = new Dictionary<string, string>
+        {
+            {"Library/PlayerDataCache/WindowsStoreApps", "UAPPlayer" },
+            {"Library/PlayerDataCache/Win", "StandalonePlayer" },
+        };
+
+        private static readonly HashSet<string> ExcludedYamlAssetExtensions = new HashSet<string> { ".jpg", ".csv", ".meta", ".pfx", ".txt", ".nuspec", ".asmdef", ".yml", ".cs", ".md", ".json", ".ttf", ".png", ".shader", ".wav", ".bin", ".gltf", ".glb", ".fbx", ".FBX", ".pdf", ".cginc" };
```

Comment:
>FBX [](start = 264, length = 3)

Nit: Do you know why FBX is the only one that is special cased where both the upper case and lower case ones are done? If we do want to generally handle upper/lower it seems like it would be okay to just store the lower version here and then to-lower the comparison strings down below where things are used.

Reply:
Good point, will do that.

---
In reply to: [277822172](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277822172) [](ancestors = 277822172)